### PR TITLE
New updates

### PR DESCRIPTION
**Pull request by KaymasJain**

### CODE REVIEW

File: contracts/receivers/aave-v2-receiver/helpers.sol [hunk 1]
```diff
@@ -104,18 +104,17 @@ abstract contract Helpers is Stores, DSMath, Variables {
...
                 uint k = j * 2;
                 if (i < num - 1) {
-                    // borrow spell
-                    castData[k] = abi.encode("6abcd3de", _token, splitAmt, 2, 0, 0); // TODO: verify this & is rate mode right?
-                    // deposit spell
-                    castData[k+1] = abi.encode("ce88b439", _token, splitAmt, 2, 0, 0); // TODO: verify this & is rate mode right?
+                    targets[k] = "AAVE-V1-A";
+                    castData[k] = abi.encodeWithSignature("borrow(address,uint256,uint256,uint256,uint256)", _token, splitAmt, 2, 0, 0);
+                    targets[k+1] = "AAVE-V1-A";
+                    castData[k+1] = abi.encodeWithSignature("deposit(address,uint256,uint256,uint256,uint256)", _token, splitAmt, 2, 0, 0);
```
Comment (thrilok209) [inline contracts/receivers/aave-v2-receiver/helpers.sol:112]:
```solidity
abi.encodeWithSignature("deposit(address,uint256,uint256,uint256)", _token, splitAmt, 0, 0);
```

Deposit func has only 4 [args](https://github.com/InstaDApp/dsa-connectors-new/blob/e93efb9226f3380b258947778f70fe7110bd2f0f/contracts/mainnet/connectors/aave/v2/main.sol#L18-L23)

Comment (KaymasJain) [reply contracts/receivers/aave-v2-receiver/helpers.sol]: Fixed!

File: contracts/senders/aave-v2-migrator/helpers.sol [hunk 2]
```diff
@@ -70,13 +70,13 @@ abstract contract Helpers is DSMath, Stores, Variables {
 
             totalBorrow[i] = add(stableBorrow[i], variableBorrow[i]);
             if (totalBorrow[i] > 0) {
-                IERC20(_token).approve(address(aave), totalBorrow[i]); // TODO: Approval is to Aave address of atokens address?
+                IERC20(_token).approve(address(aave), totalBorrow[i]);
             }
-            aave.borrow(_token, totalBorrow[i], 2, 3088, address(this)); // TODO: Borrowing debt to payback
+            aave.borrow(_token, totalBorrow[i], 2, 3088, address(this));
```
Comment (thrilok209) [inline contracts/senders/aave-v2-migrator/helpers.sol:75]:
```suggestion
            aave.borrow(_token, totalBorrow[i], 2, 3228, address(this));
```

File: contracts/senders/aave-v2-migrator/main.sol [hunk 3]
```diff
@@ -112,6 +112,26 @@ contract MigrateResolver is LiquidityResolver {
...
         );
 
+        for (uint i = 0; i < _data.supplyTokens.length; i++) {
+            address _token = _data.supplyTokens[i];
+            for (uint j = 0; j < _data.supplyTokens.length; j++) {
+                if (j != i) {
+                    require(j != i, "token-repeated");
+                }
+            }
+            require(_token != wethAddr, "should-be-eth-address");
```
Comment (thrilok209) [inline contracts/senders/aave-v2-migrator/main.sol:122]: Over here, we are converting into ETH => WETH: https://github.com/Instadapp/dsa-polygon-migration/blob/20a9e4fc6dc20e4983b70901a9a3fddef46e2c05/contracts/senders/aave-v2-connector/main.sol#L35-L36

Comment (KaymasJain) [reply contracts/senders/aave-v2-migrator/main.sol:122]: Oops! Let me set the limit for `ethAddr` then. Also, in connector, we'll need to also add for `borrowTokens`

And on Polygon does it treat WETH & ETH as 2 different tokens?

Comment (mubaris) [reply contracts/senders/aave-v2-migrator/main.sol]: On Polygon, there's only WETH since there's no ETH on Polygon

**Pull request merged**